server/src/definitions/DefinitionUtils.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Find in a text the last substring that could be a valid YML entity name's substring.
+ * @param text Some text.
+ * @returns the canditate entity's name or null if nothing could be found.
+ */
+export function getDocumentLastWord(text: string) {
+    if (!text) {
+      return null;
+    }
+    const matchArray = text.match(/\b([^:\.\s]+)\b$/g);
```

Comment:
I have no idea why I didn't mention it earlier, but… shouldn't we extract the regex to avoid compiling it multiple times? Especially now that this is in a file that already contains some regexp constants. :slightly_smiling_face: